doc/1.4/language.md

```diff
@@ -1615,6 +1609,55 @@ and inline methods remain mainly for compatibility reasons.
 In DML 1.4, methods can be `exported` using the
 [`export` declaration](#export-declarations).
 
+### Method Qualifiers
+There are a number of qualifiers that can be applied to method declarations.
+These are specified through a space delimited list immediately before the
+`method` keyword in a declaration, and any method declaration overriding another
+must share the same qualifiers as the overridden method declaration. Method
+qualifiers can be used for both regular methods as well as`shared` methods.
+
+#### Independent Methods
+Methods that do not rely on the particular instance of the device model may
+be declared `independent`:
+```
+independent method m(...) -> (...) {...}
+```
+[Exported](#export-declarations) independent methods do not have the input
+parameter corresponding to the device instance, allowing them to be called
+in greater number of contexts. The body of independent methods may not contain
+statements or expressions that rely on the device instance in any way; for
+example, `session` or `saved` variables may not be referenced, `after` and `log`
+statements may not be used, and non-`independent` may not be called.
+
+Within a template, `shared independent method`s may be declared. An example of
+this is the `model_init()` method of the
+[`model_init`](dml-builtins.html#model_init) template.
+
+#### Memoized Methods
+
+Methods can be declared `memoized` as a means of memoization: after the
+first call of an memoized method, all subsequent calls for the simulation
+session return the results of the first call without executing the body of the
+method. Memoized methods are declared as follows:
+```
+memoized method m() -> (...) {...}
+```
+Memoized methods have the restriction that they may not take any input
+parameters. If a memoized method call throws, then subsequent calls will throw
+without executing the body.
+
+Results are cached per device instance, except for memoized methods that are
+also declared independent, in which case result caching is shared across all
+device instances. This mechanism can be used to compute device-independent data
+which is then shared across all device instances of the model.
+
+For `shared` memoized methods, result caching is done per-instance, and is not
```

Comment:
You say independent memoized are shared across instances, whereas shared memoized are not. Then what about shared independent memoized methods? I would assume "shared across instances" but I can't derive that clearly from the docs.

Reply:
This is a case of two different meanings of "instance"; independent memoized are shared across _device_ instances. shared memoized are never shared across _template_ instances.
I'll try to make this more clear in the docs.